data/transactions/logic/eval_test.go

```diff
@@ -154,9 +153,48 @@ func (ep *EvalParams) reset() {
 	for i := range ep.TxnGroup {
 		ep.TxnGroup[i].ApplyData = transactions.ApplyData{}
 	}
-	ep.created = &resources{}
+	if ep.available != nil {
+		ep.available.apps = nil
+		ep.available.asas = nil
+		// reinitialize boxes because evaluation can add box refs for app creates.
+		ep.available.boxes = ep.recalcBoxRefs()
+	}
 	ep.appAddrCache = make(map[basics.AppIndex]basics.Address)
-	ep.Trace = &strings.Builder{}
+	if ep.Trace != nil {
+		ep.Trace = &strings.Builder{}
+	}
+}
+
+// reinitializeBoxRefs is nearly a copy of what happens in NewEvalParams, but is
```

Comment:
@jannotti Minor - I _think_ the method got renamed after writing the comment causing minor drift.

Alternatively - I opened https://github.com/jannotti/go-algorand/pull/11 to optionally illustrate a way around the code duplication.

```suggestion
// recalcBoxRefs is nearly a copy of what happens in NewEvalParams, but is
```

Reply:
Thanks. fixed locally to avoid the git roundtrip